packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,516 @@
+  } else if (orientation == EXOrientationLandscapeRight) {
+    return UIInterfaceOrientationMaskLandscapeRight;
+  } else {
+    return INVALID_MASK;
+  }
+}
+
+- (UIInterfaceOrientationMask)orientationLockJSToNative:(EXOrientationLock)orientationLock
+{
+  if (orientationLock == EXOrientationDefaultLock) {
+    return UIInterfaceOrientationMaskAllButUpsideDown;
+  } else if (orientationLock == EXOrientationAllLock) {
+    return UIInterfaceOrientationMaskAll;
+  } else if (orientationLock == EXOrientationPortraitLock) {
+    return UIInterfaceOrientationMaskPortrait | UIInterfaceOrientationMaskPortraitUpsideDown;
+  } else if (orientationLock == EXOrientationPortraitUpLock) {
+    return UIInterfaceOrientationMaskPortrait;
+  } else if (orientationLock == EXOrientationPortraitDownLock) {
+    return UIInterfaceOrientationMaskPortraitUpsideDown;
+  } else if (orientationLock == EXOrientationLandscapeLock) {
+    return UIInterfaceOrientationMaskLandscape;
+  } else if (orientationLock == EXOrientationLandscapeLeftLock) {
+    return UIInterfaceOrientationMaskLandscapeLeft;
+  } else if (orientationLock == EXOrientationLandscapeRightLock) {
+    return UIInterfaceOrientationMaskLandscapeRight;
+  } else if (orientationLock == EXOrientationAllButUpsideDownLock) { // legacy
+    return UIInterfaceOrientationMaskAllButUpsideDown;
+  }else {
+    return INVALID_MASK;
+  }
+}
+
+- (EXOrientation)stringToOrientation:(NSString *)orientationString
+{
+  return [[[EXScreenOrientationModule getStringToOrientationJSDict] objectForKey:orientationString] intValue];
+}
+
+- (NSString *)orientationToString:(EXOrientation)orientation
+{
+  return [[EXScreenOrientationModule getOrientationJSToStringDict] objectForKey:@(orientation)];
+}
+
+
+- (EXOrientationLock)stringToOrientationLock:(NSString *)orientationLockString
+{
+  return [[[EXScreenOrientationModule getStringToOrientationLockJSDict] objectForKey:orientationLockString] intValue];
+}
+
+- (NSString *)orientationLockToString:(EXOrientationLock)orientationLock
+{
+  return [[EXScreenOrientationModule getOrientationLockJSToStringDict] objectForKey:@(orientationLock)];
+}
+
+- (NSString *)UIInterfaceOrientationToEXOrientation:(UIInterfaceOrientation)screenOrientation
+{
+    switch (screenOrientation) {
+      case UIInterfaceOrientationPortrait:
+        return [self orientationToString:EXOrientationPortraitUp];
+      case UIInterfaceOrientationPortraitUpsideDown:
+        return [self orientationToString:EXOrientationPortraitDown];
+      case UIInterfaceOrientationLandscapeRight:
+        return [self orientationToString:EXOrientationLandscapeRight];
+      case UIInterfaceOrientationLandscapeLeft:
+        return [self orientationToString:EXOrientationLandscapeLeft];
+      default:
+        return [self orientationToString:EXOrientationUnknown];
+    }
+}
+
+- (UIInterfaceOrientation)UIDeviceOrientationToUIInterfaceOrientation:(UIDeviceOrientation)deviceOrientation
```

Comment:
Please split this file into a couple more, with helpers, conversions, having to deal with a 500-line long file is difficult